uvicorn/protocols/http/httptools_impl.py

```diff
@@ -26,7 +26,7 @@
 from uvicorn.server import ServerState
 
 HEADER_RE = re.compile(b'[\x00-\x1f\x7f()<>@,;:[]={} \t\\"]')
```

Comment:
Does the name should allow it as well, or is this good?

Reply:
I suspect that, in the case of header names, the above regular expression does not match anything that is actually valid…

I will take another look at the new HTTP RFCs

Reply:
Sorry for the late comment.

In short:

RFC 9110 has the following relevant bits:

In [Section 5.1 Field names](https://datatracker.ietf.org/doc/html/rfc9110#section-5.1):
```
  field-name     = token
```

And in [Section 5.6.1 Tokens](https://datatracker.ietf.org/doc/html/rfc9110#section-5.6.2):
```
  token          = 1*tchar

  tchar          = "!" / "#" / "$" / "%" / "&" / "'" / "*"
                 / "+" / "-" / "." / "^" / "_" / "`" / "|" / "~"
                 / DIGIT / ALPHA
                 ; any VCHAR, except delimiters
```

 * A check of the `tchar` whitelist from RFC 9110 shows that it does not overlap at all with the blacklist of the `HEADER_RE` regular expression. Thus, there is no reason for there to be an issue where someone's valid header name would be blocked by this check;
 * Only 2 ASCII characters are not covered by either the whitelist from RFC 9110, or the `HEADER_RE` blacklist: `/` and `?`. I can add them to the `HEADER_RE` regex if you would like to;
 * In principle, any byte above 0x80 would be forbidden by the standard too. Should they also be added to `HEADER_RE`?

Reply:
In other words, I think that the `HEADER_RE` regular expression for header names is fine, in that it does not block any standard-valid input.